src/script/script_scanner.cpp

```diff
@@ -52,8 +53,12 @@ bool ScriptScanner::AddFile(const char *filename, size_t basepath_length, const
 	if (!FioCheckFileExists(filename, this->subdir) || !FioCheckFileExists(this->main_script, this->subdir)) return false;
 
 	this->ResetEngine();
-	this->engine->LoadScript(filename);
-
+	try {
+		this->engine->LoadScript(filename);
+	} catch (Script_FatalError e) {
```

Comment:
The general recommendation is to catch exceptions by reference to avoid copy constructors being run during exception handling.

Reply:
I did as other catches in this area, but I guess they should all change to reference. That can be done in another PR I think.